### PR TITLE
RavenDB-17117: try to print backup status when we fail to save backup status

### PR DESCRIPTION
**Pull request by garayx**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17117

### Additional description

If we fail to save backup status on backup, we don't show the exception, also now we will try to print the backup status that save in memory (in periodic status)

### CODE REVIEW

File: src/Raven.Client/Documents/Smuggler/SmugglerResult.cs [hunk 1]
```diff
@@ -50,7 +50,16 @@ public SmugglerResult()
...
         public IOperationProgress Progress => _progress;
 
-        public IReadOnlyList<string> Messages => _messages;
+        public IReadOnlyList<string> Messages
+        {
+            get
+            {
+                lock (this)
+                {
+                    return _messages;
```
Comment (ayende) [inline src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: That is a really bad idea. You are locking, but returning the a mutable instance.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Documents/Smuggler/SmugglerResult.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: done

File: src/Raven.Client/Documents/Smuggler/SmugglerResult.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public SmugglerResult()
 
         public IOperationProgress Progress => _progress;
 
-        public IReadOnlyList<string> Messages => _messages;
+        public IEnumerable<string> Messages => _messages;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: This is a breaking change

Comment (garayx) [reply src/Raven.Client/Documents/Smuggler/SmugglerResult.cs]: reverted

**Pull request merged**